rapid7_insight_agent/plugin.spec.yaml

```
@@ -0,0 +1,191 @@
+      type: string
+      description: "Key"
+      required: false
+    value:
+      title: "Value"
+      type: string
+      description: "Value"
+      required: false
+  hostNames:
+    name:
+      title: "Name"
+      type: string
+      description: "Name"
+      required: false
+  primaryAddress:
+    ip:
+      title: "IP"
+      type: string
+      description: "IP"
+      required: false
+    mac:
+      title: "MAC"
+      type: string
+      description: "MAC address"
+      required: false
+  uniqueIdentity:
+    id:
+      title: "ID"
+      type: string
+      description: "ID"
+      required: false
+    source:
+      title: "Source"
+      type: string
+      description: "Source"
+      required: false
+  agent:
+    attributes:
+      title: "Attributes"
+      type: "[]attributes"
+      description: "Attributes"
+      required: false
+    description:
+      title: "Description"
+      type: string
+      description: "Description"
+      required: false
+    hostNames:
+      title: "Host Names"
+      type: "[]hostNames"
+      description: "Host names"
+      required: false
+    id:
+      title: "Id"
+      type: string
+      description: "Id"
+      required: false
+    primaryAddress:
+      title: "Primary Address"
+      type: primaryAddress
+      description: "Primary address"
+      required: false
+    uniqueIdentity:
+      title: "Unique Identity"
+      type: "[]uniqueIdentity"
+      description: "Unique identity"
+      required: false
+    vendor:
+      title: "Vendor"
+      type: string
+      description: "Vendor"
+      required: false
+    version:
+      title: "Version"
+      type: string
+      description: "Version"
+      required: false
+
+connection:
+  api_key:
+    title: API Key
+    description: User or Organization Key from the Insight Platform
+    type: credential_secret_key
+    example: a5zy0a6g-504e-46bz-84xx-1b3f5ci36l99
+    required: true
+  org_id:
+    title: Organization ID
+    description: Organization ID
+    type: credential_secret_key
+    example: 2cg99z6y-795n-7bzf-hj67-12355h554974
+    required: true
+  region:
+    title: Region
+    description: Region
+    type: string
+    default: United States
+    example: United States
+    required: true
+    enum:
+      - United States
+      - Europe
+      - Canada
+      - Australia
+      - Japan
+actions:
+  get_agent_details:
+    title: Get Agent Details
+    description: This action is used to find and display detailed information about a device
+    input:
+      agent:
+        title: Agent
+        description: IP address, MAC address, or host name of the device to get information from
+        type: string
+        required: true
+        example: MaliciousHost
+    output:
+      agent:
+        title: Agent
+        description: Agent information
+        type: agent
+        required: true
+  quarantine:
+    title: Quarantine
+    description: Quarantine or release quarantine on a device
```

```suggestion
    description: Quarantine or unquarantine on a device
```
Unless `release quarantine` is specific terminology the Insight Agent uses - if that is the case then we should update that terminology throughout the plugin.